nannyml/performance_estimation/direct_loss_estimation/metrics.py

```diff
@@ -92,7 +92,7 @@ def _logger(self) -> logging.Logger:
     def __str__(self):
         return self.__class__.__name__
 
-    def fit(self, reference_data: pd.DataFrame):
+    def fit(self, reference_data: pd.DataFrame, categorical_feature_columns: List[str]):
```

Comment:
I'm not very keen on breaking the paradigm we currently have. Not saying it's the best, but it is what we have for now. 

I would rather have the categorical column names list be a property of the Metric class than break our current (overly) simple interface for fitting and estimating. 

I've replaced this by introducing a property in the abstract Metric class. This can then be set during estimator fitting (since we only then know what columns are continuous or categorical).